### ISSUE TITLE
Organization: Should we merge all the "use cases" sections?

### ISSUE THREAD

**Issue by AmeliaBR** [Maps4HTML/HTML-Map-Element-UseCases-Requirements#100]
Context: I'm working on sketching out the use cases & capabilities for the [data formats/server APIS section](https://maps4html.github.io/HTML-Map-Element-UseCases-Requirements/#map-formats). And I'm finding that there I'm duplicating content from the other sections: "use a base layer from an established map provider", "create a custom tileset layer". When written as basic use cases, they're very similar to the use cases for the viewer.  Which is something we also found with the use as the use cases for client-side APIs.

At the capability level, talking about declarative features vs dynamic APIs vs data structures is relevant. But in the end, the use case is "put a map on a website with this data and these functions" and it integrates all sides.

So, I'm considering another major re-organization of the draft: **pull out all the use cases into their own top-level section.** The use cases would cover the overall goal, regardless of whether it is going to be achieved with markup or JS or files on a server.

The top-level "use cases" section could still be broken into web author use cases vs end-user use cases, as we currently do for the [map viewer section](https://maps4html.github.io/HTML-Map-Element-UseCases-Requirements/#map-viewers).  Or maybe split up even further, e.g., to separate simple website / content creator use cases (make it easy) vs web app author use cases or more advanced geospatial use cases.

Each use case would still have a list of related capabilities, but now that list could be include capabilities for the viewer, the DOM API, and the file/server structure.  We could continue to have list multiple possible ways to achieve the use case ([example](https://maps4html.github.io/HTML-Map-Element-UseCases-Requirements/#use-case-single-location): displaying a point on a basic street map requires _either_ a default streetmap base layer that comes with the widget _or_ the ability to set a tile server). With the reorganization, that would mean sometimes saying either a declarative or an API could meet the use case.

Thoughts?
Especially pinging @nchan0154 @NickFitz since you've been working on trying to write & organize the use cases & capabilities using the current structure.

**Comment by NickFitz** [comment]
My initial reaction is that this would make sense. I'm encountering situations where the overlap between the viewer and client APIs use cases is such that various chunks of text could simply be replicated from one to the other. So it seems reasonable to bring them together.

**Comment by nchan0154** [comment]
I agree this would be a good idea, like Nick I've been finding that some sections could be duplicated.

**Comment by AmeliaBR** [comment]
Ok, then I'll go ahead with the merge. If you two have any edits in progress that affect use case sections, could you try to tidy them up & submit them by your end-of-day Tuesday? Then I can do the reorganization without creating too many git conflicts later.

**Comment by nchan0154** [comment]
I don't have any in progress edits to the use case sections at the moment! 👍

**Comment by NickFitz** [comment]
I've just submitted PR #102 which is the only stuff of mine that would be directly affected by merging the use cases. Once the merge is done, I'll make any adjustments that turn out to be needed to ID references and so on in the capabilities and examples.

**Comment by AmeliaBR** [comment]
Ok, I've pushed the major reorganization, as 05fef9f  

From my commit notes:

Probably needs another clean-up pass, but this should cover the main reorganization.

Also merge all the example files, since they are tied to use cases. Still need to rename some of these & merge a couple files.

To avoid broken links, I've removed explicit entries from the "Supported use cases" section
of each capability assessment. These should be auto-generated from the "Required capabilities" sections of use cases, anyway.

Also to do: add some more capabilities for things which we had API use cases but not declarative, and vice versa.

@NickFitz I did end up moving around a couple of the API capabilities as well, to group them into two sub-categories.  Let me know if that's turning into conflicts for you & we can sort it out together. (Might want to split it into even more sub-categories, but I didn't want to do _too_ much reorg of that section without warning you.)

**Comment by NickFitz** [comment]
@AmeliaBR No conflicts at my end, as I've been keeping my drafts out of the main document for the most part. (This also means that breaking it down into further subcategories shouldn't be a problem.)

**Comment by AmeliaBR** [comment]
Follow-up: I pushed some more use-case stuff along with the reorganization of the example files (or at least, the first batch of them).

Example file names have changed, so be careful on merging/rebasing your active work.

The new example template is https://github.com/Maps4HTML/HTML-Map-Element-UseCases-Requirements/blob/master/examples/example-template.html

The first grouping of use cases (content author use cases) now all have example pages that match this template. Some of them don't have any examples in them yet, but they all match the template!  Feel free to fill in any of these.

I'm still working on reviewing & harmonizing the rest of the use cases / examples. Let me know if anyone is working on those ones.

**Comment by AmeliaBR** [comment]
PS, Heads up to @nchan0154 : Remember on Tuesday when you asked if I was still going to do any edits to the capability introductions, moving things over to the use cases? Turns out I _hadn't_ actually pushed all of that until now. Sorry. I am clearly very bad at version control. Which is to say, I'm very bad at finishing one task before getting distracted by another!

**Comment by nchan0154** [comment]
Good to know! I only had one merge conflict pulling down these changes, so crisis averted :) Will open a PR for the rest of the capability introductions today.

**Comment by AmeliaBR** [comment]
I think it's safe to close this now. Any problems can be dealt with in separate issues.
